src/app/shell/admin-tools/data/history-log/history-log-filters/history-log-filters.component.ts

```diff
@@ -1,100 +1,150 @@
-        this.filtersForm.addControl(FilterOptions.PropertyName, new FormControl(''));
+        this.addFormControlForFiltersForm(FilterOptions.PropertyName);
         break;
       case HistoryLogTypes.Users:
-        this.formControlName = FilterOptions.ShowParents;
-        this.filtersForm.addControl(FilterOptions.ShowParents, new FormControl(''));
+        this.addFormControlForFiltersForm(FilterOptions.ShowParents);
         break;
     }
   }
 
   private removeExtraFormControls(): void {
-    const extraFormControls = Object.keys(this.filtersForm.controls).filter((controlName: string) => {
-      controlName !== 'dateFrom' && controlName !== 'dateTo';
-    });
+    const extraFormControls = Object.keys(this.filtersForm.controls).filter(
+      (controlName: string) => controlName !== FilterOptions.dateFrom && controlName !== FilterOptions.dateTo
+    );
     for (const control of extraFormControls) {
       this.filtersForm.removeControl(control);
     }
   }
 
   private setBaseFiltersForm(): void {
     const currentDate = new Date();
-    const monthAgoDate = new Date(currentDate);
+    const monthAgoDate = new Date();
     monthAgoDate.setMonth(currentDate.getMonth() - 1);
 
     this.filtersForm = this.fb.group({
-      dateFrom: new FormControl(monthAgoDate),
-      dateTo: new FormControl(currentDate)
+      [FilterOptions.dateFrom]: new FormControl(monthAgoDate),
+      [FilterOptions.dateTo]: new FormControl(currentDate),
+      [this.formControlName]: new FormControl(null)
     });
   }
+
+  private emitFilterData(): void {
+    const { dateFrom, dateTo, ...restFilters } = this.filtersForm.value;
+
+    if (dateFrom && dateTo) {
+      const filtersWithEditedTime = this.setTimePeriodEqualToWholeDay(dateFrom, dateTo);
+      Object.assign(filtersWithEditedTime, restFilters);
+      this.filterData.emit(filtersWithEditedTime);
+    } else {
+      this.filterData.emit(this.filtersForm.value);
+    }
+  }
+
+  private setCustomTimeInDate(date: Date, hours: number, minutes: number, seconds: number): void {
+    date.setHours(hours);
+    date.setMinutes(minutes);
+    date.setSeconds(seconds);
+  }
+
+  private setTimeDependsOnTimezone(dateFrom: Date, dateTo: Date): FilterData {
+    const timezoneGap = dateFrom.getTimezoneOffset() * 60 * 1000;
+    const dateFromWithTimezoneGap = dateFrom.getTime() - timezoneGap;
+    const dateToWithTimezoneGap = dateTo.getTime() - timezoneGap;
+
+    return {
+      [FilterOptions.dateFrom]: new Date(dateFromWithTimezoneGap).toUTCString(),
+      [FilterOptions.dateTo]: new Date(dateToWithTimezoneGap).toUTCString()
+    };
+  }
+
+  private setTimePeriodEqualToWholeDay(dateFrom: Date, dateTo: Date): FilterData {
+    this.setCustomTimeInDate(dateFrom, 0, 0, 0);
+    this.setCustomTimeInDate(dateTo, 23, 59, 59);
+
+    return this.setTimeDependsOnTimezone(dateFrom, dateTo);
+  }
+
+  private addFormControlForFiltersForm(formControlName: string, additionalFormControlName?: string): void {
```

Comment:
You can pass an array of control names, so we don't need to add an additional parameter if we need more controls in the future

Reply:
I don't understand exactly. I have to assign a new value to the variables on 143 and 146 lines from this array, because they are used in the layout. How can I do it without editing this method in the future? I don't get it.

This is the case  how I see it, but I'm not satisfied of this

<pre>
  private addFormControlForFiltersForm(formControlsNames: string[]): void {
    formControlsNames.forEach((formControlName: string, index: number) => {
      this.filtersForm.addControl(formControlName, new FormControl(''));

      if (index === 0) {
        this.formControlName = formControlName;
      } else if (index === 1) {
        this.additionalFormControlName = formControlName;
      }
    });
  }
   </pre>
   
    What do you think?

Reply:
In my understanding, it should create a kind of config that can be used in a template.

As a result, we should get something like this 
```
{ 
  formControlName,
  options,
  order/id/etc.
} 
```

As a result, we don't need many variables to display drop-down lists in the template, just one ngFor loop will be enough.
You may need to create a different config that matches form control name and drop-down options.

As a result, we get something like this

```
 private addFormControlForFiltersForm(formControlNames: string[]): void {
    formControlNames.forEach((controlName: string) => {
      this.filtersForm.addControl(controlName, new FormControl(''));
      this.config.push({ controlName, options: this.dropdownOptionsConfig[controlName] });
    });
  }
  ```